test/e2e/autoscaler_test.go

```diff
@@ -0,0 +1,55 @@
+//go:build e2e
+// +build e2e
+
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	"k8s.io/utils/pointer"
+)
+
+var _ = Describe("When using the autoscaler with Cluster API [Autoscaler] [PR-Blocking]", func() {
+	AutoscalerSpec(ctx, func() AutoscalerSpecInput {
+		return AutoscalerSpecInput{
+			E2EConfig:              e2eConfig,
+			ClusterctlConfigPath:   clusterctlConfigPath,
+			BootstrapClusterProxy:  bootstrapClusterProxy,
+			ArtifactFolder:         artifactFolder,
+			SkipCleanup:            skipCleanup,
+			InfrastructureProvider: "docker",
+			Flavor:                 pointer.String("autoscaler"),
+			AutoscalerVersion:      "v1.26.1",
+		}
+	})
+})
+
+var _ = Describe("When using the autoscaler with Cluster API using ClusterClass [ClusterClass] [Autoscaler] [PR-Blocking]", func() {
```

Comment:
Q: Are we using the Autoscaler tag for filtering tests? If we don't foresee that we need this, let's drop the tag

For now I would like to only use tags if we actually leverage them somewhere. Just to keep the number down and managable